shoop/front/models/stored_basket.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2015, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+
+from django.conf import settings
+from django.db import models
+from shoop.core.fields import MoneyField, TaggedJSONField
+import random
+
+ID_KEYSPACE = "0123456789abcdefghijklmnopqrstuvwxyz"
+
+
+def generate_key():
+    return "".join(random.choice(ID_KEYSPACE) for x in range(32))
+
+
+class StoredBasket(models.Model):
+    # A combination of the PK and key is used to retrieve a basket for session situations.
+    key = models.CharField(max_length=32, default=generate_key)
```

Comment:
Seems that Django folks have thought this already in `django.utils.crypto.get_random_string`. See https://github.com/django/django/commit/1525874238fd705ec17a066291935a9316bd3044.